### PR TITLE
fix(webpack-dev-server): update direct dependencies

### LINKED ISSUE #43232: Missing "devServer" types on Webpack
I'm using the version `4.41.7` of the package `@types/webpack`, and I think that is missing [`devServer`](https://webpack.js.org/configuration/dev-server/) type on `Configuration` interface.

image

@tkqubo
@bumbleblym
@bcherny
@tommytroylin
@mohsen1

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- swap webpack version to v5
https://github.com/webpack/webpack-dev-server/blob/master/package.json#L106

Thanks!

Fixes #43232

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x]  Test the change in your own code. (Compile and run.)
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Review by lex111 — APPROVED** [review]

File: types/webpack-dev-server/package.json [hunk 1]
```diff
@@ -2,6 +2,8 @@
   "private": true,
   "dependencies": {
     "chokidar": "^3.5.1",
-    "http-proxy-middleware": "^2.0.0"
+    "http-proxy-middleware": "^2.0.0",
+    "tapable": "2.2.1",
```
Comment (henrikhermansen) [inline types/webpack-dev-server/package.json:6]: Why is `tapable` added here?

File: types/webpack-dev-server/package.json [hunk 2]
```diff
@@ -2,6 +2,8 @@
...
+    "webpack": "^5.54.0"
```
Comment (henrikhermansen) [inline types/webpack-dev-server/package.json:7]: Why is `webpack` added here?
In #43232 we are talking about how the dependency on `@types/webpack` causes issues, those are separate packages.

Also, the dependency on `@types/webpack` does not need to be such a high/specific version number. `"^5"` should be just fine, but even better perhaps, would be simply `"*"`, as the dev server types just add to the Webpack types.

Comment (henrikhermansen) [reply types/webpack-dev-server/package.json:7]: Maybe even better would be a peerDependency on `@types/webpack` for `"^4 | ^5"`, as I am unsure about earlier major verions.

However, that will require more work to be done when version 6 comes, and I don't know how DefinatelyTyped views peerDeps vs deps.

Do you have any thoughts on this, @lex111?

Comment (peterblazejewicz) [reply types/webpack-dev-server/package.json]: it works with '*' and without `tapable`, @henrikhermansen thx!

Comment (henrikhermansen) [reply types/webpack-dev-server/package.json]: Awesome! Thanks for taking the time to fix this annoying issue 😃 Happy to help!

**Review by henrikhermansen — APPROVED** [review]

**Pull request merged**